sdk/dnsresolver/Microsoft.Azure.Management.DnsResolver/tests/ScenarioTests/DnsResolverTests.cs

```diff
@@ -0,0 +1,643 @@
+                resourceGroupName: resourceGroupName,
+                dnsResolverName: createdDnsResolver.Name,
+                parameters: dnsResolverForCreation
+                );
+
+            updatedDnsResolver.Should().NotBeNull();
+            updatedDnsResolver.ProvisioningState.Should().BeEquivalentTo(Constants.ProvisioningStateSucceeded);
+        }
+
+        [Fact]
+        public void PatchDnsResolver_ResolveExistsAddTags_ExpectSuccess()
+        {
+            var resourceGroupName = this.CreateResourceGroup().Name;
+            var createdDnsResolver = this.CreateDnsResolver(resourceGroupName: resourceGroupName, tags: null);
+            var tagsForUpdate = TestDataGenerator.GenerateTags();
+
+            var updatedDnsResolver = this.DnsResolverManagementClient.DnsResolvers.Update(
+                resourceGroupName: resourceGroupName,
+                dnsResolverName: createdDnsResolver.Name,
+                parameters: TestDataGenerator.GenerateDnsResolver(tags: tagsForUpdate));
+
+            updatedDnsResolver.Should().NotBeNull();
+            updatedDnsResolver.ProvisioningState.Should().BeEquivalentTo(Constants.ProvisioningStateSucceeded);
+            updatedDnsResolver.Tags.Should().BeEquivalentTo(tagsForUpdate);
+        }
+
+        [Fact]
+        public void PatchDnsResolver_ResolverExistsChangeTags_ExpectSuccess()
+        {
+            var resourceGroupName = this.CreateResourceGroup().Name;
+            var createdDnsResolver = this.CreateDnsResolver(resourceGroupName: resourceGroupName, tags: TestDataGenerator.GenerateTags());
+            var tagsForUpdate = TestDataGenerator.GenerateTags();
+
+            var updatedDnsResolver = this.DnsResolverManagementClient.DnsResolvers.Update(
+                resourceGroupName: resourceGroupName,
+                dnsResolverName: createdDnsResolver.Name,
+                parameters: TestDataGenerator.GenerateDnsResolver(tags: tagsForUpdate));
+
+            updatedDnsResolver.Should().NotBeNull();
+            updatedDnsResolver.ProvisioningState.Should().BeEquivalentTo(Constants.ProvisioningStateSucceeded);
+            updatedDnsResolver.Tags.Should().BeEquivalentTo(tagsForUpdate);
+        }
+
+        [Fact]
+        public void PatchDnsResolver_ResolverExistsRemoveTags_ExpectSuccess()
+        {
+            var resourceGroupName = this.CreateResourceGroup().Name;
+            var createdDnsResolver = this.CreateDnsResolver(resourceGroupName: resourceGroupName, tags: TestDataGenerator.GenerateTags());
+
+            var updatedDnsResolver = this.DnsResolverManagementClient.DnsResolvers.Update(
+                resourceGroupName: resourceGroupName,
+                dnsResolverName: createdDnsResolver.Name,
+                parameters: TestDataGenerator.GenerateDnsResolver(tags: null));
+
+            updatedDnsResolver.Should().NotBeNull();
+            updatedDnsResolver.ProvisioningState.Should().BeEquivalentTo(Constants.ProvisioningStateSucceeded);
```

Comment:
We should ensure that other properties were not removed/changed in the process.

Reply:
Added `BeSuccessfullyUpdatedWithExpectedTags` to validate that Dns Resolver have all fields unchanged except the field updated (tags).